src/adapters/rw-adapter/src/index.ts

```diff
@@ -364,36 +329,48 @@ export default class RwAdapter implements Adapter.Service {
     return await response.json();
   }
 
-  async filterUpdate(
+  /**
+   * Serialize the editor's filters for the widgetConfig
+   * @param filters Filters
+   */
+  private getSerializedFilters(filters: any[]) {
+    const getSerializedValue = (value, type) => {
+      if (type === 'date') {
+        if (Array.isArray(value)) {
+          return value.map(date => date.toISOString());
+        }
+
+        return value.toISOString();
+      }
+
+      return value;
+    }
+
+    return filters
+      .filter(filter => filter.value !== undefined && filter.value !== null)
+      .map(({ column, type, value, notNull }) => ({
+        name: column,
+        type,
+        value: getSerializedValue(value, type),
+        notNull,
+      }));
+  }
+
+  /**
+   * Deserialize the filters for for the widget-editor's application
+   * @param filters Serialized filters
+   * @param fields Dataset's fields
+   * @param dataset Dataset object
+   */
+  async getDeserializedFilters(
     filters: any,
-    fields: any,
-    widget: Widget.Payload,
+    fields: any[],
     dataset: Dataset.Payload
-  ) {
+  ): Promise<any[]> {
```

Comment:
if possible, maybe we should define the WE filter type 

widget editor type: `Promise<Filter[]>`, it can be added here: https://github.com/Vizzuality/widget-editor/blob/master/src/packages/types/src/filters.ts


i dont think we need to define the RW one as thats out of our "control" but if we define the widget editor one we can get away with some testing for free. 

Instead of using `any` for the RW filter i like to define it something like this: 

`type ExternalFilter = any`, as it will show up better when compilation fails. and we should avoid using a `raw any` type within the application code itself, i know its all over the place atm but if we do this it will be easier to define them later. 

another note, the fields: any[] i have defined some generic types that you can use: https://github.com/Vizzuality/widget-editor/blob/master/src/packages/types/src/generic.ts we can add more here if its needed, for example `type GenericArray = any[]`